### ISSUE TITLE
Add option to limit `find_next_char` and siblings to one line

### ISSUE THREAD

**Issue by BrianShTsoi** [helix-editor/helix#11312]
`f`/`t` are mostly used for not too distant selection or navigation, especially since `gw` is available now. I think it is not common to have a situation where we want to use `f`/`t`move through multiple lines (feel free to give counter examples to prove me wrong)? 
The fact that it keeps on searching through the file makes it less forgiving. I admit this is skill issue on my part, but often times I typed the wrong character or press `Alt-.` too many times and it overshoots down some number of lines and I have no idea where I ended up in the file. 

I understand this is a matter of preference. Would it be possible to have the option to limit `f` and siblings to a single line (such as in the form of a separate command so that we can remap it)?

**Comment by Aqaao** [comment]
Agree, sometimes I accidentally type in the wrong char and then cursor jumps hundreds of lines away. I had to type `alt-;`,`;` and `zz` to get back. The worst part is that if I accidentally cancel selection, I need to manually jump back to original position.

I think we need a setting to toggle between full text move and single line move. Adding an extra one-line version of each command is too redundant.  We can via `:toggle single_line_move.enble` command to toggle it.
